File: app/src/main/java/com/voipgrid/vialer/util/LoginRequiredActivity.java
package com.voipgrid.vialer.util;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;

import com.voipgrid.vialer.Preferences;
import com.voipgrid.vialer.logging.Logger;
import com.voipgrid.vialer.logging.VialerBaseActivity;
import com.voipgrid.vialer.onboarding.OnboardingActivity;

/**
 * Super class to handle logged in state of activity.
 */
public abstract class LoginRequiredActivity extends VialerBaseActivity {

    private AccountHelper mAccountHelper;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mAccountHelper = new AccountHelper(this);
    }

    @Override
    protected void onResume() {
        super.onResume();

        Preferences prefs = new Preferences(this);
        if (!prefs.isLoggedIn() && prefs.finishedOnboarding()) {
            new Logger(LoginRequiredActivity.class).w("Not logged in anymore! Redirecting to onboarding");
            // Go to onboarding.
            Intent intent = new Intent(new Intent(this, OnboardingActivity.class));
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
            finish();
        }
    }

    protected String getEmail() {
        return mAccountHelper.getEmail();
    }

    protected String getPassword() {
        return mAccountHelper.getPassword();
    }

    protected String getApiToken() {
        return mAccountHelper.getApiToken();
    }

    protected boolean hasApiToken() {
        return mAccountHelper.hasApiToken();
    }
}

File: app/src/main/java/com/voipgrid/vialer/util/LoginRequiredActivity.kt
package com.voipgrid.vialer.util

import android.content.Intent
import com.voipgrid.vialer.User
import com.voipgrid.vialer.api.UserSynchronizer
import com.voipgrid.vialer.logging.Logger
import com.voipgrid.vialer.logging.VialerBaseActivity
import com.voipgrid.vialer.onboarding.OnboardingActivity
import org.koin.android.ext.android.inject

abstract class LoginRequiredActivity : VialerBaseActivity() {

    override val logger = Logger(this)

    protected val userSynchronizer: UserSynchronizer by inject()

    override fun onResume() {
        super.onResume()

        if (!User.isLoggedIn) {
            logger.w("Not logged in anymore! Redirecting to onboarding")
            startActivity(Intent(this, OnboardingActivity::class.java).apply {
                flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            })
            finish()
        }
    }
}